pkg/registry/apps/statefulset/strategy_test.go

```diff
@@ -22,76 +22,100 @@ import (
-	}
-	if Strategy.AllowCreateOnUpdate() {
-		t.Errorf("StatefulSet should not allow create on update")
-	}
+	testFn := func(t *testing.T) {
+		ctx := genericapirequest.NewDefaultContext()
+		if !Strategy.NamespaceScoped() {
+			t.Errorf("StatefulSet must be namespace scoped")
+		}
+		if Strategy.AllowCreateOnUpdate() {
+			t.Errorf("StatefulSet should not allow create on update")
+		}
 
-	validSelector := map[string]string{"a": "b"}
-	validPodTemplate := api.PodTemplate{
-		Template: api.PodTemplateSpec{
-			ObjectMeta: metav1.ObjectMeta{
-				Labels: validSelector,
+		validSelector := map[string]string{"a": "b"}
+		validPodTemplate := api.PodTemplate{
+			Template: api.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: validSelector,
+				},
+				Spec: api.PodSpec{
+					RestartPolicy: api.RestartPolicyAlways,
+					DNSPolicy:     api.DNSClusterFirst,
+					Containers:    []api.Container{{Name: "abc", Image: "image", ImagePullPolicy: "IfNotPresent"}},
+				},
 			},
-			Spec: api.PodSpec{
-				RestartPolicy: api.RestartPolicyAlways,
-				DNSPolicy:     api.DNSClusterFirst,
-				Containers:    []api.Container{{Name: "abc", Image: "image", ImagePullPolicy: "IfNotPresent"}},
+		}
+		ps := &apps.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
+			Spec: apps.StatefulSetSpec{
+				PodManagementPolicy: apps.OrderedReadyPodManagement,
+				Selector:            &metav1.LabelSelector{MatchLabels: validSelector},
+				Template:            validPodTemplate.Template,
+				UpdateStrategy:      apps.StatefulSetUpdateStrategy{Type: apps.RollingUpdateStatefulSetStrategyType},
+				PersistentVolumeClaimPolicy: &apps.StatefulSetPersistentVolumeClaimPolicy{
+					OnSetDeletion: apps.DeletePersistentVolumeClaimDeletePolicyType,
+					OnScaleDown:   apps.RetainPersistentVolumeClaimDeletePolicyType,
+				},
 			},
-		},
-	}
-	ps := &apps.StatefulSet{
-		ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
-		Spec: apps.StatefulSetSpec{
-			PodManagementPolicy: apps.OrderedReadyPodManagement,
-			Selector:            &metav1.LabelSelector{MatchLabels: validSelector},
-			Template:            validPodTemplate.Template,
-			UpdateStrategy:      apps.StatefulSetUpdateStrategy{Type: apps.RollingUpdateStatefulSetStrategyType},
-		},
-		Status: apps.StatefulSetStatus{Replicas: 3},
-	}
+			Status: apps.StatefulSetStatus{Replicas: 3},
+		}
 
-	Strategy.PrepareForCreate(ctx, ps)
-	if ps.Status.Replicas != 0 {
-		t.Error("StatefulSet should not allow setting status.replicas on create")
-	}
-	errs := Strategy.Validate(ctx, ps)
-	if len(errs) != 0 {
-		t.Errorf("unexpected error validating %v", errs)
-	}
+		Strategy.PrepareForCreate(ctx, ps)
+		if ps.Status.Replicas != 0 {
+			t.Error("StatefulSet should not allow setting status.replicas on create")
+		}
+		errs := Strategy.Validate(ctx, ps)
+		if len(errs) != 0 {
+			t.Errorf("unexpected error validating %v", errs)
+		}
```

Comment:
check that if feature gate is disabled, `PeristentVolumeClaimPolicy` is nil.

Reply:
done